comp/api/api/apiimpl/internal/agent/agent_test.go

```diff
@@ -63,3 +70,447 @@ func TestSetupHandlers(t *testing.T) {
+			method: "POST",
+			payload: installinfo.SetInstallInfoRequest{
+				Tool:             "",
+				ToolVersion:      "1.0",
+				InstallerVersion: "ecs-task-def-v1",
+			},
+			expectedStatus: http.StatusBadRequest,
+			expectedError:  true,
+		},
+		{
+			name:   "missing tool version field",
+			method: "POST",
+			payload: installinfo.SetInstallInfoRequest{
+				Tool:             "ECS",
+				ToolVersion:      "",
+				InstallerVersion: "ecs-task-def-v1",
+			},
+			expectedStatus: http.StatusBadRequest,
+			expectedError:  true,
+		},
+		{
+			name:   "missing installer version field",
+			method: "POST",
+			payload: installinfo.SetInstallInfoRequest{
+				Tool:             "ECS",
+				ToolVersion:      "1.0",
+				InstallerVersion: "",
+			},
+			expectedStatus: http.StatusBadRequest,
+			expectedError:  true,
+		},
+	}
+
+	router := setupRoutes()
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var body []byte
+			var err error
+			if tt.payload != nil {
+				if str, ok := tt.payload.(string); ok {
+					body = []byte(str)
+				} else {
+					body, err = json.Marshal(tt.payload)
+					require.NoError(t, err)
+				}
+			}
+
+			fullURL, err := url.JoinPath(ts.URL, "/install-info")
+			require.NoError(t, err)
+
+			req, err := http.NewRequest(tt.method, fullURL, bytes.NewReader(body))
+			require.NoError(t, err)
+			req.Header.Set("Content-Type", "application/json")
+
+			resp, err := ts.Client().Do(req)
+			require.NoError(t, err)
+			defer resp.Body.Close()
+
+			assert.Equal(t, tt.expectedStatus, resp.StatusCode)
+
+			bodyBytes, err := io.ReadAll(resp.Body)
+			require.NoError(t, err)
+
+			if tt.expectedError {
+				if tt.expectedStatus == http.StatusMethodNotAllowed {
+					// Router rejected the request; body likely empty or plain text
+					assert.Empty(t, bodyBytes)
+					return
+				}
+
+				var response installinfo.SetInstallInfoResponse
+				err = json.Unmarshal(bodyBytes, &response)
+				require.NoError(t, err)
+				assert.False(t, response.Success)
+				assert.NotEmpty(t, response.Message)
+			} else {
+				var response installinfo.SetInstallInfoResponse
+				err = json.Unmarshal(bodyBytes, &response)
+				require.NoError(t, err)
+				assert.True(t, response.Success)
+				assert.NotEmpty(t, response.Message)
+
+				getURL, err := url.JoinPath(ts.URL, "/install-info")
+				require.NoError(t, err)
+
+				getReq, err := http.NewRequest("GET", getURL, nil)
+				require.NoError(t, err)
+
+				getResp, err := ts.Client().Do(getReq)
+				require.NoError(t, err)
+				defer getResp.Body.Close()
+
+				var info installinfo.InstallInfo
+				err = json.NewDecoder(getResp.Body).Decode(&info)
+				require.NoError(t, err)
+				assert.Equal(t, "ECS", info.Tool)
+				assert.Equal(t, "1.0", info.ToolVersion)
+				assert.Equal(t, "ecs-task-def-v1", info.InstallerVersion)
+			}
+		})
+	}
+}
+
+func TestHandleGetInstallInfo(t *testing.T) {
```

Comment:
Same comment as above. We can probably merge these assertions in the TestInstallInfoAPIRoutes test function 😄